msprime/simulations.py

```diff
@@ -1471,6 +1489,198 @@ def print_history(self, output=sys.stdout):
+        """
+        This function will calculate the mean coalescence rates between the
+        lineages of the sample configuration provided in `num_samples`, at
+        each generation time listed by `steps`, in this demographic model.
+        The coalescence rate at time t in the past is the average rate
+        of coalescence of as-yet-uncoalesed lineages, computed as follows:
+        let p(t) be the probability that the lineages of a randomly
+        chosen pair of samples has not yet coalesced by time t,
+        let p(z,t) be the probability that the lineages of a randomly
+        chosen pair of samples has not yet coalesced by time t *and* are
+        both in population z, and let N(z,t) be the diploid effective
+        population size of population z at time t. Then the mean coalescence
+        rate at time t is r(t) = (sum_z p(z,t) / (2 * N(z,t) ) / p(t).
+        The computation is done approximating population size trajetories
+        with piecewise constant trajectories, in steps of size dt = end / num_steps.
+        For this to be accurate, dt must be small enough so that (a) short epochs
+        (e.g., bottlenecks) are not missed, and (b) populations do not
+        change in size too much over that time, if they are growing or shrinking.
+        If in doubt, check that results are simliar after doubling `num_steps`.
+
+        :param list steps: The respective time steps (in generations) for which the
+            coalescence rate, as well as probablility that a randomly pair of lineages
+            has not yet coalesced, is computed and returned. The list given should be
+            sorted in ascending order.
+
+        :param list num_samples: is a list of the same length as the number
+            of populations, so that `num_samples[j]` is the number of sampled
+            chromosomes in subpopulation `j`.
+
+        :param int min_pop_size: is the smallest allowed population size during
+            computation of coalescent rates (i.e., coalescence rates are actually
+            1 / 2 * max(min_pop_size, N(z,t))). Spurious very small population sizes
+            can occurIn models where populations grow exponentially but are unused
+            before some time in the past, and lead to floating point error.
+            This should be set to a value smaller than the smallest
+            desired population size in the model.
+
+        :param bool double_step_validatation: provides the option to ensure your step
+            sizes are small enough to ensure the correctness of a model's coalescent
+            rate prediction. If `True` this will, at most, take four times
+            the computation. Note: This test is sesitive to small effective
+            population sizes. Currently, this check with throw a warning
+            if the relative difference, for all values of coalecence rate or
+            sum of P statistic exceed 1e-3.
```

Comment:
How about
:param bool double_step_validation: Whether to perform the check that step sizes are sufficiently small, as described above. This is highly recommended, and will take at most four times the computation.